wandb/sdk/launch/agent/agent.py

```diff
@@ -0,0 +1,177 @@
+            )
+        except Exception as e:
+            print("Exception:", e)
+            return None
+        return ups
+
+    def print_status(self) -> None:
+        """Prints the current status of the agent."""
+        print(
+            "polling on project {}, queues {} for jobs".format(
+                self._project, " ".join(self._queues)
+            )
+        )
+
+    def finish_job_id(self, job_id: int) -> None:
+        """Removes the job from our list for now."""
+        # TODO:  keep logs or something for the finished jobs
+        del self._jobs[job_id]
+        self._running -= 1
+
+    def _update_finished(self, job_id: int) -> None:
+        """Check our status enum."""
+        if self._jobs[job_id].get_status().state in ["failed", "finished"]:
+            self.finish_job_id(job_id)
+
+    def run_job(self, job: Dict[str, Any]) -> None:
+        """Sets up project and runs the job."""
+        # TODO: logger
+        print("agent: got job", job)
+        # parse job
+        launch_spec = job["runSpec"]
+        if launch_spec.get("overrides") and isinstance(
+            launch_spec["overrides"].get("args"), list
+        ):
+            launch_spec["overrides"]["args"] = util._user_args_to_dict(
+                launch_spec["overrides"].get("args", [])
+            )
+        project = create_project_from_spec(launch_spec, self._api)
+        project = fetch_and_validate_project(project, self._api)
+
+        resource = launch_spec.get("resource") or "local"
+        backend_config: Dict[str, Any] = {
+            PROJECT_DOCKER_ARGS: {},
+            PROJECT_SYNCHRONOUS: True,
+        }
+        if _is_wandb_local_uri(self._base_url):
+            if sys.platform == "win32":
+                backend_config[PROJECT_DOCKER_ARGS]["net"] = "host"
+            else:
+                backend_config[PROJECT_DOCKER_ARGS]["network"] = "host"
+            if sys.platform == "linux" or sys.platform == "linux2":
+                backend_config[PROJECT_DOCKER_ARGS][
+                    "add-host"
+                ] = "host.docker.internal:host-gateway"
+
+        backend_config["runQueueItemId"] = job["runQueueItemId"]
+        backend = load_backend(resource, self._api, backend_config)
+        backend.verify()
+
+        run = backend.run(project)
+        if run:
+            self._jobs[run.id] = run
+            self._running += 1
+
+    def loop(self) -> None:
+        """Main loop function for agent."""
+        wandb.termlog(
+            "launch agent polling project {}/{} on queues: {}".format(
+                self._entity, self._project, ",".join(self._queues)
+            )
+        )
+        try:
+            while True:
+                self._ticks += 1
+                for queue in self._queues:
+                    job = self.pop_from_queue(queue)
+                    if job:
+                        break
+                if not job:
+                    time.sleep(30)
```

Comment:
Maybe 10 seconds would be better here?  The queue request is very inexpensive, I could imagine someone launching something and being confused that nothing happened for 30 seconds.

Reply:
That sounds good. I'm also replacing this with a constant rather than hard coding